src/controller/CHIPDevice.cpp

```diff
@@ -457,16 +464,20 @@ CHIP_ERROR Device::LoadSecureSessionParameters(ResetTransport resetNeeded)
         SuccessOrExit(err);
     }
 
-    err = mSessionManager->NewPairing(Optional<Transport::PeerAddress>::Value(mDeviceAddress), mDeviceId, &pairingSession,
-                                      SecureSession::SessionRole::kInitiator, mAdminId);
-    SuccessOrExit(err);
+    if (IsOperationalCertProvisioned())
+    {
+        err = WarmupCASESession();
+        SuccessOrExit(err);
+    }
+    else
+    {
+        err = pairingSession.FromSerializable(mPairing);
+        SuccessOrExit(err);
 
-    // TODO - Enable CASE Session setup before message is sent to a fully provisioned device
-    // if (IsProvisioningComplete())
-    // {
-    //     err = EstablishCASESession();
-    //     SuccessOrExit(err);
-    // }
+        err = mSessionManager->NewPairing(Optional<Transport::PeerAddress>::Value(mDeviceAddress), mDeviceId, &pairingSession,
+                                          SecureSession::SessionRole::kInitiator, mAdminId);
```

Comment:
Are we potentially making life harder with the AdminId keys?  I understand that these 16-bit numbers are concise.  But the spec gives us other options: fabric reference is only 64-bit, and fabric index is only 8-bit.

Why the need for yet a third flavor of key?  Also, the fact that the spec formalizes fabric index as an 8-bit number implies that there is a hard-limit of 256 fabrics that a Matter node can be a member of.  Why then the internal use of a 16-bit Admin ID?  I don't think we can actually support a 16-bit space here.

In any case, if a 16-bit admin ID does make sense, it is hard to understand why this would ever be seen outside of the `AdminPairingTable` object.